src/environment/stella_environment.cpp

```diff
@@ -195,6 +203,26 @@ bool StellaEnvironment::isTerminal() const {
      m_state.getEpisodeFrameNumber() >= m_max_num_frames_per_episode));
 }
 
+void StellaEnvironment::pressSelect(size_t num_steps) {
+  Event* event = m_osystem->event();
+  m_state.pressSelect(event);
+  for (size_t t = 0; t < num_steps; t++) {
+    m_osystem->console().mediaSource().update();
+  }
+  processScreen();
+  processRAM();
+  emulate(PLAYER_A_NOOP, PLAYER_B_NOOP);
+}
+
+void StellaEnvironment::setDifficulty(difficulty_t mask) {
+  Event* event = m_osystem->event();
+  m_state.setDifficulty(event, mask);
+  m_osystem->console().mediaSource().update();
+  processScreen();
+  processRAM();
+  m_difficulty = mask;
```

Comment:
Is there a reason for story m_difficulty in StellaEnvironment, rather than ALEState?

Reply:
Well, I was using it inside `StellaEnvironment::reset()`. I moved it to `ALEState` and I dropped the line `setDifficulty(m_difficulty);`. I believe I don't have to set `m_difficulty` inside `ALEState` anymore since it is now in `ALEState`. When I reset the `StellaEnvironment` it will be preserved since it will be already in `ALEState`.

@mgbellemare do you agree?